### ISSUE TITLE
Windows compatibility

### ISSUE THREAD

**Issue by D0ve** [Exalm/gnomerunnergtk-revived#6]
Thank you for so wonderful SM theme!
Would be great to see it working on win systems either!
Could you remove linux dependency in the theme?
...or include windows compatibility, please!

**Comment by Exalm** [comment]
I have an idea that would totally allow this (along with merging stock GTK and FD flavors back into one, and switching themes on the fly), but it will limit compatibility to 2.28+.
AMO statistics say that old SM versions are used by literally one-two people, so I think I can drop them.

The idea is replacing all usages of native icons with variables and making several css files containing only those variables, which will then be loaded with JS.
Like that, I can make several layers:
1. Few base files that contain all used icons. Those would work usable even in non-Unix environments.
2. Stock GTK "layer". It is meant to be used in combination with one of the above.
3. Freedesktop "layer". Same as above, excluding GTK one.
So, it will be like:
Icon theme: ...
Native icons: None/GTK/Freedesktop/(Possibly)KDE.
Like that all the icon usage will be in a single file and easily controlled.

Like this:
Base file:
--icon-go-previous-16:       url("chrome://icon-themes/gnome/16x16/go-previous.png");
--icon-go-previous-16-dis: var(--icon-go-previous-16);
--icon-go-previous-24:       url("chrome://icon-themes/gnome/24x24/go-previous.png");
--icon-go-previous-24-dis: var(--icon-go-previous-24);
--icon-go-previous-48:       url("chrome://icon-themes/gnome/48x48/go-previous.png");
--icon-go-previous-48-dis: var(--icon-go-previous-48);
--icon-go-previous-dis-opacity: .5;

GTK layer:
--icon-go-previous-16:       url("moz-icon://stock/gtk-go-back?size=menu");
--icon-go-previous-16-dis: url("moz-icon://stock/gtk-go-back?size=menu&state=disabled");
--icon-go-previous-24:       url("moz-icon://stock/gtk-go-back?size=toolbar");
--icon-go-previous-24-dis: url("moz-icon://stock/gtk-go-back?size=toolbar&state=disabled");
--icon-go-previous-48:       url("moz-icon://stock/gtk-go-back?size=dialog");
--icon-go-previous-48-dis: url("moz-icon://stock/gtk-go-back?size=dialog&state=disabled");
--icon-go-previous-dis-opacity: 1;

Freedesktop layer:
--icon-go-previous-16:       url("moz-icon://stock/go-previous?size=menu");
--icon-go-previous-16-dis: url("moz-icon://stock/go-previous?size=menu&state=disabled");
--icon-go-previous-24:       url("moz-icon://stock/go-previous?size=toolbar");
--icon-go-previous-24-dis: url("moz-icon://stock/go-previous?size=toolbar&state=disabled");
--icon-go-previous-48:       url("moz-icon://stock/go-previous?size=dialog");
--icon-go-previous-48-dis: url("moz-icon://stock/go-previous?size=dialog&state=disabled");
--icon-go-previous-dis-opacity: 1;

This approach works well, as far as I tested.

**Comment by dumol** [comment]
I would add that this theme shouldn't be Linux only. IIRC I had to install it in Linux and then move the profile to the BSD installation I was using at the time to get it working there too. I think all the BSDs could benefit from installing this… And I can confirm it works beatifully in FreeBSD and OpenBSD. Thanks!

**Comment by Exalm** [comment]
The platform restriction is entirely addons.mozilla.org thing. There aren't any ways to disable it for non-GTK SeaMonkey builds, you are only offered by this:
![default](https://cloud.githubusercontent.com/assets/1510457/8312608/092f68f8-19f7-11e5-99cb-6a6d59d0602d.png)

**Comment by Exalm** [comment]
Of course, if I implement this, I will be able to remove the restriction altogether.

**Comment by Exalm** [comment]
Also, you could just download the theme even though it says to be incompatible and then force install it. The theme itself isn't restricted, and that's not even possible, as far as I know.

**Comment by dumol** [comment]
Thanks for the tips, I'll keep that in mind.

**Comment by Exalm** [comment]
Hmm. Icons are the least problem, is seems. With XP Luna theme it mostly looks acceptable, but in Windows Classic theme it's _completely_ broken.
SeaMonkey Unix skin and GNOMErunner rely mostly on -moz-appearance for widgets. However, Classic does not support it, so toolbars are borderless, toolbar buttons are borderless, menu popups are transparent etc.

**Comment by Exalm** [comment]
I've finally merged the slowly developed config-rewrite branch. The icon themes are there, and they'll work on Windows as well. I won't allow Windows platform by default on AMO for now, but the theme itself should work on non-classic themes. On windows classic, it will be a mess because moz-appearance is not supported there, see my previous comment.
